apps/web/src/components/ExperienceFormFields/WorkFields/GovFields.tsx

```diff
@@ -212,26 +210,14 @@ const GovFields = ({ labels }: SubExperienceFormProps) => {
     if (watchGovEmploymentType !== WorkExperienceGovEmployeeType.Contractor) {
       resetDirtyField("govContractorRoleSeniority");
       resetDirtyField("govContractorType");
-    }
-
-    if (
-      watchGovContractorType === GovContractorType.SelfEmployed ||
-      watchGovEmploymentType !== WorkExperienceGovEmployeeType.Contractor
-    ) {
-      resetField("contractorFirmAgencyName", {
-        keepDirty: false,
-        defaultValue: undefined,
-      });
```

Comment:
This shouldn't be removed entirely
Contractors can be self-employed or agency, but agency is the only one with a name field. So switching from agency to self-employed should also wipe the name field but not reset the other contractor fields

Reply:
Okay, this was adding back in [revert agency name condition, remove redundant type condition](https://github.com/GCTC-NTGC/gc-digital-talent/pull/12884/commits/7f3a40104d3f00380959ad7df96ff8060b423f78)